grpc/rpclog/grpc.go

```diff
@@ -0,0 +1,95 @@
+// Copyright © 2017 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package rpclog
+
+import (
+	"time"
+
+	"github.com/TheThingsNetwork/go-utils/log"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// ServerOptions for logging RPCs
+func ServerOptions(log log.Interface) []grpc.ServerOption {
+	return []grpc.ServerOption{
+		grpc.UnaryInterceptor(UnaryServerInterceptor(log)),
+		grpc.StreamInterceptor(StreamServerInterceptor(log)),
+	}
+}
+
+// ClientOptions for logging RPCs
+func ClientOptions(log log.Interface) []grpc.DialOption {
+	return []grpc.DialOption{
+		grpc.WithUnaryInterceptor(UnaryClientInterceptor(log)),
+		grpc.WithStreamInterceptor(StreamClientInterceptor(log)),
+	}
+}
+
+// UnaryServerInterceptor logs unary RPCs on the server side
+func UnaryServerInterceptor(log log.Interface) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		log := log.WithField("method", info.FullMethod)
+		log = log.WithFields(FieldsFromContext(ctx))
+		start := time.Now()
+		resp, err = handler(ctx, req)
```

Comment:
Because Unary calls should return immediately, and we don't want to log two lines within milliseconds for a single request